offchain/authority-claimer/src/checker.rs

```diff
@@ -83,32 +93,53 @@ impl DuplicateChecker for DefaultDuplicateChecker {
     type Error = DuplicateCheckerError;
 
     async fn is_duplicated_rollups_claim(
-        &self,
+        &mut self,
         dapp_address: Address,
         rollups_claim: &RollupsClaim,
     ) -> Result<bool, Self::Error> {
-        let proof_context =
-            U256([rollups_claim.epoch_index, 0, 0, 0]).encode().into();
-
-        match self
-            .authority
-            .get_claim(H160(dapp_address.inner().to_owned()), proof_context)
-            .block(ethers::types::BlockNumber::Latest)
-            .call()
+        self.update_claims(dapp_address).await?;
+        Ok(self.claims.iter().any(|read_claim| {
+            &read_claim.epoch_hash == rollups_claim.epoch_hash.inner()
+                && read_claim.first_index == rollups_claim.first_index
+                && read_claim.last_index == rollups_claim.last_index
+        }))
+    }
+}
+
+impl DefaultDuplicateChecker {
+    async fn update_claims(
+        &mut self,
+        dapp_address: Address,
+    ) -> Result<(), DuplicateCheckerError> {
+        let current_block = self
+            .provider
+            .get_block_number()
             .await
-        {
-            // If there's any response, the claim already exists
-            Ok(_response) => Ok(true),
-            Err(e) => {
-                // If there's an InvalidClaimIndex error, we're asking for an index
-                // bigger than the current one, which means it's a new claim
-                if e.to_string().contains("InvalidClaimIndex()") {
-                    Ok(false)
-                } else {
-                    info!("{:?}", e);
-                    Err(e).context(ContractSnafu)
-                }
-            }
+            .context(ProviderSnafu)?
+            .as_u64();
```

Comment:
This will not work properly if a reorg happens. With the state-fold code, we looked N blocks behind to be sure. We should do the same here.

Reply:
Just pushed a fix that should solve this issue